### PR TITLE
chore: refine `@rolldown/core` package

### PR DESCRIPTION
**Pull request by hyf0**
### Description

- `@rolldown/core` will emit correct files in dist to expose to outside
  - This is only a temp solution for building packages. I need to play around with rolldown as a dependecy, and the old 
otuion is not usable. 

### Test Plan

---

### CODE REVIEW

**Comment by hyf0** [comment]
Current dependencies on/for this PR:
* main
  * **PR #219** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/219?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #220** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/220?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #221** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/221?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
        * **PR #223** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/223?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #224** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/224?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/rolldown-rs/rolldown/221?utm_source=stack-comment).

File: web/playground/package.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "description": "create an app to consume rust-generated wasm packages",
   "main": "index.js",
   "scripts": {
-    "build": "vite build",
```
Comment (hyf0) [inline web/playground/package.json:7]: Tracked in https://github.com/rolldown-rs/rolldown/issues/222.

File: packages/core/tsconfig.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "extends": "../../tsconfig.json",
   "compilerOptions": {
-    "composite": true,
+    // "composite": true,
```
Comment (hyf0) [inline packages/core/tsconfig.json:4]: 🤷 Don't know why. https://stackoverflow.com/questions/60029058/project-must-list-all-files-or-use-an-include-pattern

**Review by underfin — APPROVED** [review]

**Comment by hyf0** [comment]
### Merge activity

* **Nov 11, 1:40 AM**: @hyf0 started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/rolldown-rs/rolldown/221).
* **Nov 11, 1:41 AM**: [Graphite](https://app.graphite.dev/github/pr/rolldown-rs/rolldown/221) rebased this pull request as part of a merge.
* **Nov 11, 1:42 AM**: @hyf0 merged this pull request with [Graphite](https://app.graphite.dev/github/pr/rolldown-rs/rolldown/221).

**Pull request merged**